api.ts

```diff
@@ -1683,15 +1686,15 @@ export interface BaseFormulaDef<ParamDefsT extends ParamDefs, ResultT extends st
 export type StringFormulaDef<ParamDefsT extends ParamDefs> = BaseFormulaDef<ParamDefsT, string> & {
   resultType: ValueType.String;
   execute(params: ParamValues<ParamDefsT>, context: ExecutionContext): Promise<string> | string;
-} & {schema?: StringSchema; codaType?: StringHintTypes};
```

Comment:
moved these to FormulaOptions. The only other usage of StringFormulaDef and NumericFormulaDef is:
```
const def: StringFormulaDef<ParamDefsT> & {codaType?: StringHintTypes; formulaSchema?: StringSchema} = {
        ...fullDefinition,
        codaType: 'codaType' in fullDefinition ? fullDefinition.codaType : undefined,
        formulaSchema: 'schema' in fullDefinition ? fullDefinition.schema : undefined,
        validateParameters: wrapMetadataFunction(fullDefinition.validateParameters),
      };
      const {onError: _, resultType: unused, codaType, formulaSchema, ...rest} = def;
```
which means we union codaType and do not need schema there anyways